packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -284,6 +335,75 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupRef {
         };
     }
   }
+
+  private ec2TagSet(tagSet?: InstanceTagSet):
+      cloudformation.DeploymentGroupResource.EC2TagSetProperty | undefined {
+    if (!tagSet || tagSet.instanceTagGroups.length === 0) {
+      return undefined;
+    }
+
+    return {
+      ec2TagSetList: tagSet.instanceTagGroups.map(tagGroup => {
+        return {
+          ec2TagGroup: this.tagGroup2TagsArray(tagGroup) as
+            cloudformation.DeploymentGroupResource.EC2TagFilterProperty[],
```

Comment:
Why do you need the `as` here? Just make sure `tagGroup2TagsArray` indicates the return type

Reply:
Because you need to distinct types here (`EC2TagFilterProperty` for EC2 instance tags, `TagFilterProperty` for on-premise instance tags), while the structure of them is the same - so, I'm taking advantage of TypeScript's structural typing to reduce code duplication here.